### PR TITLE
Add process orchestrator wrapper support to sqllogictest

### PR DESCRIPTION
**Pull request by antiguru**
This change adds a --orchestrator-process-wrapper parameter to sqllogictest inspired by the parameter of the same name to `environmentd`. This allows for wrapping calls to `clusterd` with anther command, such as a profiler.

Also send a SIGTERM to child processes before killing to give the wrapper a moment to cleanup its state, or write down pending data to disk. This makes it possible to use `perf` as a wrapper.

For example, use the following command to run the child process with `valgrind`:
```
bin/sqllogictest --release --no-default-features -- \
  --orchestrator-process-wrapper 'valgrind --leak-check=yes  --main-stacksize=67108864' \
  test/sqllogictest/cluster.slt
```

(`--no-default-features` is important because valgrind doesn't work with jemalloc, which gets disabled by the flag.)

Related: #18114 

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/orchestrator-process/src/lib.rs [hunk 1]
```diff
@@ -716,6 +716,14 @@ async fn spawn_process(
 
     impl Drop for KillOnDropChild {
         fn drop(&mut self) {
+            if let Some(pid) = self.0.id().and_then(|id| i32::try_from(id).ok()) {
+                let _ = nix::sys::signal::kill(
+                    nix::unistd::Pid::from_raw(pid),
+                    nix::sys::signal::Signal::SIGTERM,
+                );
+                // Give the process a bit of time to react to the signal
+                std::thread::sleep(Duration::from_millis(500));
```
Comment (benesch) [inline src/orchestrator-process/src/lib.rs:725]: This is going to synchronously block on an async thread, which is not great. I think a `tokio::task::block_in_place` is in order.

**Review by benesch — COMMENTED** [review]
LGTM modulo the blocking on an async thread!

File: src/orchestrator-process/src/lib.rs [hunk 2]
```diff
@@ -716,6 +716,14 @@ async fn spawn_process(
 
     impl Drop for KillOnDropChild {
         fn drop(&mut self) {
+            if let Some(pid) = self.0.id().and_then(|id| i32::try_from(id).ok()) {
+                let _ = nix::sys::signal::kill(
```
Comment (benesch) [inline src/orchestrator-process/src/lib.rs:720]: I'm kind of conflicted about this. I consider it a feature that the process orchestrator SIGKILLs processes when they're done. I wouldn't want someone adding an "onexit" handler to `clusterd` for example. If it's easier, might be nice to only run this logic when a process wrapper is in use.

**Review by def- — APPROVED** [review]
Very useful, thanks!

File: src/orchestrator-process/src/lib.rs (hunk 1 above)
Comment (antiguru) [reply src/orchestrator-process/src/lib.rs]: Agree+thanks, TIL :)

**Pull request merged**